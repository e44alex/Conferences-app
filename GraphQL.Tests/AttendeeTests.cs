using Backend.Common.Data;
using GraphQL.Common.Types;
using GraphQL.Common.Types.Mutations;
using GraphQL.Common.Types.Queries;
using HotChocolate;
using HotChocolate.Execution;
using HotChocolate.Execution.Configuration;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;

namespace GraphQL.Tests
{
    public class AttendeeTests
    {
        private IRequestExecutorBuilder GetRequestExecutorBuilder()
        {
            return new ServiceCollection()
                .AddPooledDbContextFactory<ApplicationDbContext>(
                    options => options.UseInMemoryDatabase("Data Source=conferences.db"))
                .AddGraphQL()
                .AddQueryType(d => d.Name("Query"))
                .AddTypeExtension<AttendeeQueries>()
                .AddMutationType(d => d.Name("Mutation"))
                .AddTypeExtension<AttendeeMutations>()
                .AddType<AttendeeType>()
                .AddType<SessionType>()
                .AddType<SpeakerType>()
                .AddType<TrackType>()
                .AddGlobalObjectIdentification();
        }

        [Fact]
        public async Task Attendee_Schema_Changed()
        {
            ISchema schema = await GetRequestExecutorBuilder().BuildSchemaAsync();

            // assert
            schema.Print().MatchSnapshot();
        }

        [Fact]
        public async Task RegisterAttendee()
        {
            // arrange
            IRequestExecutor executor = await GetRequestExecutorBuilder().BuildRequestExecutorAsync();

            // act
            IExecutionResult result = await executor.ExecuteAsync(@"
            mutation RegisterAttendee {
                registerAttendee(
                    input: {
                        emailAddress: ""michael@chillicream.com""
                            firstName: ""michael""
                            lastName: ""staib""
                            userName: ""michael3""
                        })
                {
                    attendee {
                        id
                    }
                }
            }");

            // assert
            result.ToJson().MatchSnapshot();
        }
    }
}