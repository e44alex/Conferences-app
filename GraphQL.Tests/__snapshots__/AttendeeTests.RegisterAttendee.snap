{
  "errors": [
    {
      "message": "Unexpected Execution Error",
      "locations": [
        {
          "line": 3,
          "column": 17
        }
      ],
      "path": [
        "registerAttendee"
      ]
    }
  ]
}
